platform-hub-api/app/services/encryptor_service.rb

```diff
@@ -0,0 +1,15 @@
+class EncryptorService
+  def initialize(secret_key_base)
+    @encryptor = ActiveSupport::MessageEncryptor.new(secret_key_base)
+  end
+
+  def encrypt(plaintext)
+    @encryptor.encrypt_and_sign(plaintext)
+  end
+
+  def decrypt(encrypted_data)
+    @encryptor.decrypt_and_verify(encrypted_data)
+  rescue ActiveSupport::MessageVerifier::InvalidSignature
+    nil
```

Comment:
I don't think we need to. Unless we want to log every failed decryption when data didn't have valid signature?